griptape/tools/google_drive/tool.py

```diff
@@ -0,0 +1,153 @@
+            ): str,
+            Optional(Literal(
+                "max_files",
+                description="maximum number of files to return"
+            )): int
+        })
+    })
+    def _build_client(self, scopes: list[str], values: dict) -> object:
+        from google.oauth2 import service_account
+        from googleapiclient.discovery import build
+        credentials = service_account.Credentials.from_service_account_info(
+            self.service_account_credentials, scopes=scopes
+        )
+        delegated_credentials = credentials.with_subject(values["drive_owner_email"])
+        service = build('drive', 'v3', credentials=delegated_credentials)
+        return service
+
+    def list_files(self, params: dict) -> ListArtifact | ErrorArtifact:
+
+        values = params["values"]
+
+        try:
+            service = self._build_client(self.LIST_FILES_SCOPES, values)
+            results = service.files().list(pageSize=values.get('max_files', 10)).execute()
+            items = results.get('files', [])
+            return items
+
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"error retrieving files from Drive: {e}")
+
+    @activity(config={
+        "description": "Can be used to upload a file to Google Drive",
+        "schema": Schema({
+            Literal(
+                "drive_owner_email",
+                description="email of the Google Drive's owner"
+            ): str,
+            Literal(
+                "file_path",
+                description="path to the file to be uploaded"
+            ): str,
+            Literal(
+                "file_name",
+                description="name to give the file on Google Drive"
+            ): str,
+            Literal(
+                "mime_type",
+                description="MIME type of the file being uploaded"
+            ): str
+        })
+    })
+    def upload_file(self, params: dict) -> InfoArtifact | ErrorArtifact:
+        from googleapiclient.http import MediaFileUpload
+
+        values = params['values']
+
+        try:
+            service = self._build_client(self.UPLOAD_FILE_SCOPES, values)
+            file_metadata = {'name': values['file_name']}
+            media = MediaFileUpload(values['file_path'], mimetype=values['mime_type'])
+            file = service.files().create(body=file_metadata, media_body=media, fields='id').execute()
+
+            return InfoArtifact(f'A file was successfully uploaded with ID {file.get("id")}.')
+
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"error uploading file to Drive: {e}")
+
+    @activity(config={
+        "description": "Can be used to download a file from Google Drive",
+        "schema": Schema({
+            Literal(
+                "drive_owner_email",
+                description="email of the Google Drive's owner"
+            ): str,
+            Literal(
+                "file_id",
```

Comment:
We should support direct file downloading like we do in `FileManager.load_files_from_disk`. There are two options here:

1. Make this an `schema.Or` field and let the LLM download the file by `file_id` or from `paths`.
2. Add a separate activity that closely matches `FileManager.load_files_from_disk`.

I think option 2 makes more sense as it's less ambiguous for the LLM but let me know what you think.